### ISSUE TITLE
Support urllib3 v1.23

### ISSUE THREAD

**Issue by ahstro** [docker/compose#6008]
## Description of the issue
[urllib3](https://github.com/urllib3/urllib3/) released version [1.23](https://github.com/urllib3/urllib3/releases/tag/1.23) on June 5th, and it breaks docker-compose.

## Context information (for bug reports)

```
docker-compose version 1.21.2, build unknown
```

```
Client:
 Version:      18.05.0-ce
 API version:  1.37
 Go version:   go1.10.2
 Git commit:   f150324782
 Built:        Wed May 16 22:27:45 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.05.0-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.10.2
  Git commit:   f150324782
  Built:        Wed May 16 22:28:17 2018
  OS/Arch:      linux/amd64
  Experimental: false

```

## Steps to reproduce the issue

1. Upgrade urllib3
2. Run docker-compose

### Observed result
An error is thrown, saying docker-compose depends on urllib3 version `<1.23`

### Expected result
docker-compose to work as usual

**Comment by venthur** [comment]
Weird, after updating the `requirements.txt` to `urllib 1.32` and re-installing docker-compose it works again, but somehow the tests still fetch `urllib 1.22` from somewhere (and thus make the tests fail). I really can't find where it comes from -- `git grep urllib` shows only 2 lines...

**Comment by venthur** [comment]
Here's [some info](https://github.com/requests/requests/issues/4673#issuecomment-395069592) from the `requests` maintainer. We can expect version 2.20, which has updated dependency on urllib3, in the next week. I've prepared a PR which can be merged then (but fails now as requests version 2.20 doesn't exist yet).

**Comment by shin-** [comment]
We pin the version of urllib (and other dependencies) for the same reason the requests maintainers do. This is a feature, not a bug.

**Comment by shin-** [comment]
Fixed by #6092
